TestFailedStartupExitCode fails during make test-short

## Bug Report
**What did you do?**
I've tried to compile the `release-2.6.0`branch using go1.11.4 and noticed a `make test-short` failure:

**What did you expect to see?**
All test should pass.

**What did you see instead? Under which circumstances?**

`TestFailedStartupExitCode` fails during `make test-short`. All other tests pass.

```
$ make test-short
>> running short tests
GO111MODULE=on go test -short  -mod=vendor ./...
--- FAIL: TestFailedStartupExitCode (0.00s)
    main_test.go:173: unable to retrieve the exit status for prometheus: fork/exec : no such file or directory
FAIL
FAIL    github.com/prometheus/prometheus/cmd/prometheus 0.017s
...
```
AFAICS the root cause is that `promPath` is empty as it is not set in `TestMain()` for short tests:

https://github.com/prometheus/prometheus/blob/dbd1d58c894775c0788470944b818cc724f550fb/cmd/prometheus/main_test.go#L34-L43

https://github.com/prometheus/prometheus/blob/dbd1d58c894775c0788470944b818cc724f550fb/cmd/prometheus/main_test.go#L160-L175

**Environment**

* System information:
```
$ cat /etc/redhat-release
CentOS Linux release 7.5.1804 (Core)
$ uname -srm
Linux 3.10.0-862.14.4.el7.x86_64 x86_64
```

* Prometheus version:
```
$ ./prometheus --version
prometheus, version 2.6.0 (branch: release-2.6, revision: dbd1d58c894775c0788470944b818cc724f550fb)
  build user:       knweiss@mangan
  build date:       20190104-11:38:56
  go version:       go1.11.4
$ git status
On branch release-2.6
Your branch is up to date with 'origin/release-2.6'.

nothing to commit, working tree clean
```

--- a/cmd/prometheus/main_test.go
+++ b/cmd/prometheus/main_test.go
@@ -163,6 +163,10 @@ func TestComputeExternalURL(t *testing.T) {
 
 // Let's provide an invalid configuration file and verify the exit status indicates the error.
 func TestFailedStartupExitCode(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
 	fakeInputFile := "fake-input-file"
 	expectedExitStatus := 1
 

